### PR TITLE
ENH: Adds alignment API and wmrb_alignment_loss

### PR DESCRIPTION
**Pull request by jfkirk**
1. Adds alignment (cosine similarity) prediction and API method.
2. Adds an implementation of WMRB loss that optimizes for alignment.
3. Adds tests for alignment.
4. Various other small refactors.

CC @jcauteru @mesanders

### CODE REVIEW

File: tensorrec/recommendation_graphs.py [hunk 1]
```diff
@@ -44,6 +54,19 @@ def prediction_serial(tf_user_representation, tf_item_representation, tf_x_user,
...
+    """
+    Returns the cosine distance between the given user representations and item representations.
+    This is a form of dense prediction.
+    :param tf_user_representation:
+    :param tf_item_representation:
+    :return:
+    """
+    normalized_users = tf.nn.l2_normalize(tf_user_representation, 0)
+    normalized_items = tf.nn.l2_normalize(tf_item_representation, 0)
+    return tf.matmul(normalized_users, normalized_items, transpose_b=True)
```
Comment (jcauteru) [inline tensorrec/recommendation_graphs.py]: maths :+1:

**Comment by jcauteru** [comment]
"Adds an implementation of WMRB loss that optimizes for alignment." == adding wrapper for WMRB that uses cosine sim as scoring function correct? Seems that way
